modules/gentoo/2023.lua.core

```diff
@@ -0,0 +1,93 @@
+local arch, cpu_vendor_id, interconnect, cuda_driver_version=...
+
+help([[Gentoo prefix is Gentoo Linux installed in a prefix - Homepage: https://wiki.gentoo.org/wiki/Project:Prefix]])
+whatis("Gentoo prefix is Gentoo Linux installed in a prefix - Homepage: https://wiki.gentoo.org/wiki/Project:Prefix")
+
+add_property(   "lmod", "sticky")
+
+local root = "/cvmfs/soft.computecanada.ca/gentoo/2023/x86-64-v3"
+
+prepend_path("PATH", pathJoin(root, "usr/bin"))
+prepend_path("PATH", pathJoin(root, "usr/sbin"))
+prepend_path("PATH", "/cvmfs/soft.computecanada.ca/custom/bin")
+prepend_path("INFOPATH", pathJoin(root, "usr/share/binutils-data/x86_64-pc-linux-gnu/2.40/info:"))
+prepend_path("MANPATH", pathJoin(root, "usr/share/man"))
+prepend_path("MANPATH", pathJoin(root, "usr/share/binutils-data/x86_64-pc-linux-gnu/2.40/man"))
+prepend_path("MANPATH", pathJoin(root, "usr/share/gcc-data/x86_64-pc-linux-gnu/12/man"))
+prepend_path("PYTHONPATH", "/cvmfs/soft.computecanada.ca/custom/python/site-packages")
+local posix = require "posix"
+-- RedHat based systems
+if posix.stat("/etc/pki/tls/certs/","type") == "directory" then
+	setenv("SSL_CERT_FILE", "/etc/pki/tls/certs/ca-bundle.crt")
+	setenv("CURL_CA_BUNDLE", "/etc/pki/tls/certs/ca-bundle.crt")
+-- Debian based systems
+elseif posix.stat("/etc/ssl/certs/","type") == "directory" then
+	setenv("SSL_CERT_FILE", "/etc/ssl/certs/ca-certificates.crt")
+	setenv("CURL_CA_BUNDLE", "/etc/ssl/certs/ca-certificates.crt")
+end
+
+pushenv("LESSOPEN", "|lesspipe %s")
+if os.getenv("XDG_DATA_DIRS") then
+	prepend_path("XDG_DATA_DIRS", pathJoin(root, "usr/share"))
+else
+	setenv("XDG_DATA_DIRS", pathJoin(root, "usr/share") .. ":/usr/local/share:/usr/share")
+end
+if os.getenv("XDG_CONFIG_DIRS") then
+	prepend_path("XDG_CONFIG_DIRS", pathJoin(root, "etc/xdg"))
+else
+	setenv("XDG_CONFIG_DIRS", pathJoin(root, "etc/xdg") .. ":/etc/xdg")
+end
+
+require("os")
+-- Define RSNT variables
+-- do something only at load time, if not already defined. Otherwise, unloading the module will lose the predefined value if there is one
+if mode() == "load" then
+	setenv("RSNT_ARCH", arch)
+	setenv("RSNT_INTERCONNECT", interconnect)
+	setenv("RSNT_CUDA_DRIVER_VERSION", cuda_driver_version)
+	setenv("RSNT_CPU_VENDOR_ID", cpu_vendor_id)
+end
+
+-- let pip use our wheel house
+if arch == "avx512" then
```

Comment:
Is `gentoo` really `gentoo` for pip now ? i.e. how many wheels in the current `gentoo` wheelhouse may break because the 2023 version does not have the same packages as the 2020 one ?

Reply:
yes mostly this was just a copy paste with 2020 replaced by 2023.

Actually maybe better to move the pip logic to the python modules? It could be like openmpi configs with an include file?